Zookeeper Path handling

benzekrimaha opened the pull request:
Issue : BB-542

bert-e commented:
# Hello benzekrimaha,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-542 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `8.7.0`


Please check the `Fix Version/s` of BB-542, or the target
branch of this pull request.

benzekrimaha committed:
ZookeeperUrl path handling for both S3C and Artesca

Changes: the zookeeperUrl was adapted based on
the config to differenciate between S3C and Artesca
please note that a new parameter has been added to ease
the unit testing on the function , it has the default
value of the zookeeperManager class.
Issue : BB-542

benzekrimaha committed:
Add unit test coverage for _setZookeeper

Issue : BB-542

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

williamlardier commented on tests/unit/api/BackbeatAPI.spec.js:
```diff
@@ -210,4 +220,40 @@ describe('BackbeatAPI', () => {
+
+    describe('_setZookeeper', () => {
+        let zkManagerArgs;
+        class MockZookeeperManager {
+            constructor(url, options, logger) {
+                zkManagerArgs = { url, options, logger };
+                this.once = sinon.stub().callsFake((event, callback) => {
+                    if (event === 'connected') {
+                        callback();
+                    }
+                });
+                this.removeAllListeners = sinon.stub();
+            }
+        }
```
I think we should define this class outside of a `describe`

williamlardier commented on tests/unit/api/BackbeatAPI.spec.js:
```diff
@@ -210,4 +220,40 @@ describe('BackbeatAPI', () => {
+        });
+
+        it('should append zookeeperPath to connectionString if this._queuePopulator.mongo does not exist', done => {
+            delete bbapi._queuePopulator.mongo;
```
As we should not rely on the order ot unit tests for them to pass, I think this should be reverted at the end of the test (or just mock the value)

welansari commented:
Actually i think there's nothing to do here, as Zookeeper Client in 7.x is only used for heathchecks, meaning that it's only used to check whether or not Zookeeper is up and running and that we are able to connect to it. BackbeatAPI doesn't actually store any state in it. So i think we can safely leave the new path we use in 8.x.

bert-e commented:
# Conflict




There is a conflict between your branch `improvement/BB-542-zookeeper-handling` and the
destination branch `development/8.7`.

Please resolve the conflict on **the feature branch** (`improvement/BB-542-zookeeper-handling`).

```sh
git fetch && \
git checkout origin/improvement/BB-542-zookeeper-handling && \
git merge origin/development/8.7
```

Resolve merge conflicts and commit

```sh
git push origin HEAD:improvement/BB-542-zookeeper-handling
```
